pkg/controller/periodic/periodic.go

```diff
@@ -101,15 +105,31 @@ func (c *Controller) Gather() {
 }
 
 func (c *Controller) periodicTrigger(stopCh <-chan struct{}) {
```

Comment:
note to myself... we should really stop using stopCh and use context.Context :-)

/cc @iNecas 
(sounds like a good intern job :-)